_dev/src/ts/pages/UpdatePageUpdate.ts

```diff
@@ -105,4 +110,19 @@ export default class UpdatePageUpdate extends UpdatePage {
 
     await api.post(form.dataset.routeToSubmit!);
   };
+
+  #enableExitConfirmation = (): void => {
+    window.addEventListener('beforeunload', this.#handleBeforeUnload);
+  };
+
+  #disableExitConfirmation = (): void => {
+    window.removeEventListener('beforeunload', this.#handleBeforeUnload);
+  };
+
+  #handleBeforeUnload = (event: Event): void => {
+    event.preventDefault();
+
+    // Included for legacy support, e.g. Chrome/Edge < 119
+    event.returnValue = true;
+  };
```

Comment:
what do you think to export this function inside new `domUtils` to be reusable on other script without duplication ? Or add the methods inside PageAbstract ?

Reply:
This will actually be useful in the backup and restore processes. +1

Reply:
There is already a topic about duplicated code started in https://github.com/PrestaShop/autoupgrade/pull/1076, the objective here was to unblock your PR.